DB/src/main/java/io/deephaven/db/tables/lang/DBLanguageParser.java

```diff
@@ -1576,9 +1576,10 @@ public Class visit(MethodCallExpr n, VisitArgs printer) {
     }
 
     private void checkPyNumbaVectorizedFunc(MethodCallExpr n, Expression[] expressions, Class[] expressionTypes) {
-        if (n.getParentNode() != null) {
+        if (n.getParentNode() != null && (n.getParentNode().getClass() != CastExpr.class || n.getParentNode().getParentNode() != null)) {
```

Comment:
Can we have a comment here explaining what this logic is trying to do and why it is the right thing?